nemo/collections/nlp/models/dialogue/dialogue_gpt_classification_model.py

```diff
@@ -224,47 +267,72 @@ def predict_step(self, batch, batch_idx, dataloader_idx=None):
         # return self(batch)
         raise NotImplementedError()
 
+    def on_train_end(self):
+        if self.prompt_learning:
+            self.language_model.on_train_end()
+
+    def get_prompt_token_labels_for_megatron_gpt(self, input_ids, num_prompt_tokens):
+
+        prompt_token_labels = torch.full(
+            size=(input_ids.size(0), num_prompt_tokens),
+            fill_value=self.tokenizer.tokenizer.pad_token_id,
+            dtype=torch.long,
+        )
+
+        if self.prompt_learning:
+            prompt_token_labels.data = torch.LongTensor(
+                np.tile(np.array(self.language_model.pseudo_token_ids), (input_ids.size(0), 1))
+            )
+
+        prompt_token_labels = prompt_token_labels.to(input_ids.device)
+
+        return prompt_token_labels
+
+    def get_prompt_ids_for_megatron_gpt(self, input_ids):
```

Comment:
Can this method name be changed to `get_virtual_prompt_ids_for_megatron_gpt`? You have a method called `get_prompt_token_labels_for_megatron_gpt` where `prompt_token` seems to refer to all prompt tokens, whether they are virtual or discrete prompt tokens. I think making the distinction that this method is just for virtual prompt tokens would add clarity.

Reply:
Makes sense!